### PR TITLE
Added Insert position in java

### PR DESCRIPTION
**Pull request by Nidhideveloper**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added Insert position in java

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4292

### CODE REVIEW

**Comment by Nidhideveloper** [comment]
@HarshCasper Plz check the PR, thank u sir!

File: Java/other/InsertPosition.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Enter the key value to be inserted: 2
+Enter the elements for array in sorted order: [1, 4, 5, 8, 9]
+Output--
+Position at which the key value should be inserted is: 1
+
+TIME COMPLEXITY--> O(N)
+SPACE COMPLEXITY--> O(1)
+
+*/
+
```
Comment (atarax665) [inline Java/other/InsertPosition.java]: Remove this extra line.

File: Java/other/InsertPosition.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            else
+            {
+                rightptr=mid-1;
+            }
+        }
+
+        return leftptr;
+    }
+
+    //DRIVER METHOD
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
    //DRIVER CODE
```

File: Java/other/InsertPosition.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+it would best fit or be inserted in the order.
+*/
+
+import java.util.*;
+
+public class InsertPosition
+{
+    public int searchPosition(int[] arr, int key)
+    {
+        int leftptr=0;
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
        int leftptr = 0;
```

File: Java/other/InsertPosition.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        int rightptr=arr.length-1;
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
        int rightptr = arr.length - 1;
```

File: Java/other/InsertPosition.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+        while(leftptr < rightptr)
+        {
+            int mid = (leftptr+rightptr)/2;
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
            int mid = (leftptr + rightptr) / 2;
```

File: Java/other/InsertPosition.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            if(arr[mid]==key)
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
            if(arr[mid] == key)
```

File: Java/other/InsertPosition.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                return mid;
+            }
+            else if(arr[mid]>key)
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
            else if(arr[mid] > key)
```

File: Java/other/InsertPosition.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                leftptr=mid+1;
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
                leftptr = mid + 1;
```

File: Java/other/InsertPosition.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+            }
+            else
+            {
+                rightptr=mid-1;
```
Comment (atarax665) [inline Java/other/InsertPosition.java]:
```suggestion
                rightptr = mid - 1;
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Nidhideveloper** [comment]
@atarax665 Sir I have modified the code, please check.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**